react-app/src/components/ListItem/index.js

```diff
@@ -0,0 +1,87 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import ActionIcon from 'components/ActionIcon';
+import Button from 'components/Button';
+
+import styles from './index.module.scss';
+
+export default class ListItem extends React.PureComponent {
+  static propTypes = {
+    /**
+     * The date of the entry
+     */
+    date: PropTypes.string.isRequired,
+
+    /**
+     * The text of the entry
+     */
+    text: PropTypes.string.isRequired,
+
+    /**
+     * The function to call when edit button is tapped
+     */
+    onEdit: PropTypes.func.isRequired,
+
+    /**
+     * The function to call when delete button is tapped
+     */
+    onDelete: PropTypes.func.isRequired,
+  };
+
+  constructor(props) {
```

Comment:
If you don't need the props for your initial state and you don't need to do anything special, there's no reason to have a `constructor`.

Instead of:
```js
  constructor(props) {
    super(props);

    this.state = { expanded: false }
  }

  _handleEdit = () => this.props.onEdit(this.props.date);
```

You can just write:
```js
  state = { expanded: false }

  _handleEdit = () => this.props.onEdit(this.props.date);
```

Reply:
Aha! That's great!